tests/socket.rs

```diff
@@ -154,26 +154,43 @@ fn set_nonblocking() {
     assert_nonblocking(&socket, false);
 }
 
-#[test]
-fn default_flags() {
-    let socket = Socket::new(Domain::IPV4, Type::STREAM, None).unwrap();
+fn assert_default_flags(socket: &Socket, default_flags: bool) {
```

Comment:
Nit: can you change this to `assert_common_flags` to match `set_common_flags` naming, maybe even add a reference to it as comment.

Reply:
Also change `default_flags` to `expected`, since `default` wouldn't make sense anymore.